src/main/ui/chat-ui.gd

```diff
@@ -0,0 +1,266 @@
+	preload("res://assets/ui/chat/texture/bg08.png"),
+	preload("res://assets/ui/chat/texture/bg09.png"),
+	preload("res://assets/ui/chat/texture/bg10.png"),
+	preload("res://assets/ui/chat/texture/bg11.png"),
+	preload("res://assets/ui/chat/texture/bg12.png"),
+	preload("res://assets/ui/chat/texture/bg13.png"),
+	preload("res://assets/ui/chat/texture/bg14.png"),
+	preload("res://assets/ui/chat/texture/bg15.png")
+]
+
+# The dialog is copied into multiple labels. We display the smallest label which fits.
+onready var _chat_labels := {
+	ChatSize.SMALL: $Holder/ChatLabelS,
+	ChatSize.MEDIUM: $Holder/ChatLabelM,
+	ChatSize.LARGE: $Holder/ChatLabelL,
+	ChatSize.EXTRA_LARGE: $Holder/ChatLabelXl,
+}
+
+# The name is copied into multiple labels. We display the smallest label which fits.
+onready var _nametag_labels := {
+	NametagSize.SMALL: $Holder/ChatRect/NametagRect/NametagLabelS,
+	NametagSize.MEDIUM: $Holder/ChatRect/NametagRect/NametagLabelM,
+	NametagSize.LARGE: $Holder/ChatRect/NametagRect/NametagLabelL,
+	NametagSize.EXTRA_LARGE: $Holder/ChatRect/NametagRect/NametagLabelXL,
+	NametagSize.EXTRA_EXTRA_LARGE: $Holder/ChatRect/NametagRect/NametagLabelXXL
+}
+
+# Bigger chat windows are displayed using bigger textures
+onready var _chat_rect_textures := {
+	ChatSize.SMALL: preload("res://assets/ui/chat/window-small-sheet.png"),
+	ChatSize.MEDIUM: preload("res://assets/ui/chat/window-medium-sheet.png"),
+	ChatSize.LARGE: preload("res://assets/ui/chat/window-large-sheet.png"),
+	ChatSize.EXTRA_LARGE: preload("res://assets/ui/chat/window-extra-large-sheet.png")
+}
+
+# There are two nametag textures; one for small names, and four for big names
+onready var _nametag_rect_textures := {
+	NametagSize.SMALL: preload("res://assets/ui/chat/nametag-small-sheet.png"),
+	NametagSize.MEDIUM: preload("res://assets/ui/chat/nametag-large-sheet.png"),
+	NametagSize.LARGE: preload("res://assets/ui/chat/nametag-large-sheet.png"),
+	NametagSize.EXTRA_LARGE: preload("res://assets/ui/chat/nametag-large-sheet.png"),
+	NametagSize.EXTRA_EXTRA_LARGE: preload("res://assets/ui/chat/nametag-large-sheet.png")
+}
+
+func _ready() -> void:
+	_hide_labels()
+
+
+func _process(delta: float) -> void:
+	_total_time += delta
+	
+	# stretch the chat window vertically/horizontally in a circular way which preserves its area
+	$Holder/ChatRect.scale.x = 0.20 + PULSE_AMOUNT.x * sin((_total_time + 8.96) * 2 * PI / PULSE_PERIOD)
+	$Holder/ChatRect.scale.y = 0.20 + PULSE_AMOUNT.y * cos((_total_time + 8.96) * 2 * PI / PULSE_PERIOD)
+
+
+"""
+Animates the chat UI to gradually reveal the specified text, mimicking speech.
+"""
+func play_text(name: String, text_with_pauses: String, accent_def: Dictionary) -> void:
+	set_text(text_with_pauses)
+	set_name(name)
+	set_accent_def(accent_def)
+	reshape()
+
+
+"""
+Assigns the chat label's text and updates our _chat_size field to the smallest chat label which fit.
+"""
+func set_text(text_with_pauses: String) -> void:
```

Comment:
Change `text_with_pauses` to `text_with_lulls`

Reply:
Changed to `text`, the idea of lull characters seems pretty in the weeds and should stay close to ChatLabel